astropy/nddata/utils.py

```diff
@@ -25,6 +28,16 @@
     "Cutout2D",
 ]
 
+if TYPE_CHECKING:
+    from numbers import Number
+    from typing import Literal, TypeAlias
+
+    from typing_extensions import TypeVar
+
+    T = TypeVar("T")
+    SingleOrTuple: TypeAlias = T | tuple[T, ...]
+    ModeT: TypeAlias = Literal["partial", "trim", "strict"]
```

Comment:
Do you think it would be a good idea to use [`Enum`](https://docs.python.org/3.10/library/enum.html#creating-an-enum) for this? (in a separate pull request)

Reply:
Internally yes, but it shouldn't affect the user visible interface, hence the type hint should be stable. I'm curating a list of such Enum opportunities, I'll add it there.

Reply:
In this pull request `ModeT` is causing the documentation build to fail because Sphinx is trying to add a link to it in the function signatures where it is used as an annotation, but it can't find its definition. Any type aliases defined in a `if TYPE_CHECKING:` block and used in public functions will cause such failures. If the modes are not listed in an `Enum` yet then it would be best to write them out in each signature that needs them.

Reply:
damn I somehow missed that comment...